tc-server/src/main/java/com/tc/objectserver/entity/ManagedEntityImpl.java

```diff
@@ -976,16 +976,28 @@ private void getEntity(ServerEntityRequest getEntityRequest, ResultCapture respo
     if (this.isDestroyed) {
       response.failure(ServerException.createNotFoundException(getID()));
     } else {
-      if (canDelete) {
-        clientReferenceCount += 1;
-        Assert.assertTrue(clientReferenceCount > 0);
-      }
       // The FETCH can only come directly from a client so we can down-cast.
       ClientID clientID = getEntityRequest.getNodeID();
       ClientDescriptorImpl descriptor = new ClientDescriptorImpl(clientID, getEntityRequest.getClientInstance());
       boolean added = clientEntityStateManager.addReference(descriptor, this.fetchID);
+      
+      if (canDelete) {
+        if (added) {
+          clientReferenceCount += 1;
+          Assert.assertTrue(clientReferenceCount > 0);
+        } else {
+          // this should never happen.  Ther server is expecting sane things from the client. 
+          logger.info("the client has attempted to fetch the same entity instance twice {}", descriptor);
```

Comment:
If it should never happen should it be a warn?

Reply:
I debated.  Was thinking of a certain consumer that considered new warn level logging an error in tests.